src/main/scala/intellij/haskell/intention/AddParensIntention.scala

```diff
@@ -19,21 +19,15 @@ class AddParensIntention extends PsiElementBaseIntentionAction {
         end.getParent.addAfter(right, end)
       }
 
-      if (psiElement.getText.length == 1 && start.isInstanceOf[PsiWhiteSpace]) {
-        addParens(end, end)
-      } else if (psiElement.getText.length == 1 && end.isInstanceOf[PsiWhiteSpace]) {
-        addParens(start, start)
-      } else {
-        addParens(start, end)
-      }
+      addParens(start, end)
     }
   }
 
   override def isAvailable(project: Project, editor: Editor, psiElement: PsiElement): Boolean = {
     HaskellPsiUtil.getSelectionStartEnd(psiElement, editor) match {
       case None => false
       case Some((start, end)) =>
-        psiElement.isWritable && start.getNode.getElementType != HS_LEFT_PAREN && end.getNode.getElementType != HS_RIGHT_PAREN
+        psiElement.isWritable && (start.getNode.getElementType != HS_LEFT_PAREN || end.getNode.getElementType != HS_RIGHT_PAREN)
```

Comment:
It will wrap parenthesis around the identifier which the cursor is currently pointed to. I just tried last night.